DependencyInjection/Configuration.php

```diff
@@ -30,6 +30,9 @@ public function getConfigTreeBuilder()
                     ->info('The priority of the event listener that converts an Exception to a Response')
                     ->defaultValue(10)
                 ->end()
+                ->scalarNode('custom_role')
+                    ->info('Custom role that replaces \'ROLE_OAUTH2_\' role prefix')
```

Comment:
Created a factory that is injected into the services where the `OAuth2Token` constructor was called so that the token will be created in this factory.

Also rebased and retargeted